packages/yoroi-extension/chrome/extension/ergo-connector/api.js

```diff
@@ -262,13 +266,36 @@ export async function connectorSignTx(
   const jsonBoxesToSign = utxosToSign.map(formatUtxoToBox);
   processBoxesForSigmaRust(jsonBoxesToSign);
   const txBoxesToSign = RustModule.SigmaRust.ErgoBoxes.from_boxes_json(jsonBoxesToSign);
+  console.log('data inputs');
+  const dataInputs = new RustModule.SigmaRust.DataInputs();
+  for (const dataInput of tx.dataInputs) {
+    const boxId = RustModule.SigmaRust.BoxId.from_str(dataInput.boxId);
+    dataInputs.add(new RustModule.SigmaRust.DataInput(boxId));
+  }
+  // We could modify the best block backend to return this information for the previous block
+  // but I'm guessing that votes of the previous block isn't useful for the current one
+  // and I'm also unsure if any of these 3 would impact signing or not.
+  // Maybe version would later be used in the ergoscript context?
+  const headerJson = JSON.stringify({
+    version: 2, // TODO: where to get version? (does this impact signing?)
+    parentId: bestBlock.hash,
+    timestamp: Date.now(),
+    nBits: 682315684511744, // TODO: where to get difficulty? (does this impact signing?)
+    height: bestBlock.height + 1,
+    votes: "040000", // TODO: where to get votes? (does this impact signing?)
+  });
+  console.log(`block header: ${headerJson}`);
+  const blockHeader = RustModule.SigmaRust.BlockHeader.from_json(headerJson);
+  console.log('pre-header');
+  const preHeader = RustModule.SigmaRust.PreHeader.from_block_header(blockHeader);
+  console.log('before signing');
   const signedTx = RustModule.SigmaRust.Wallet
     .from_secrets(wasmKeys)
     .sign_transaction(
-      RustModule.SigmaRust.ErgoStateContext.dummy(), // TODO? Not implemented in sigma-rust
+      new RustModule.SigmaRust.ErgoStateContext(preHeader),
       wasmTx,
       txBoxesToSign,
-      RustModule.SigmaRust.ErgoBoxes.from_boxes_json([]), // TODO: not supported by sigma-rust
+      RustModule.SigmaRust.ErgoBoxes.from_boxes_json([]),
```

Comment:
I think the connector doesn't rely on [utxoTransactions.js](https://github.com/Emurgo/yoroi-frontend/blob/develop/packages/yoroi-extension/app/api/ergo/lib/transactions/utxoTransaction.js#L469), right?

Reply:
It uses `generateKeys()` but not `signTransaction()` so we don't need to worry about that TODO there for the connector.